source/bao_hyp/config.rst

```diff
@@ -3,7 +3,78 @@ Bao Configuration File
 
 Overview
 --------
-An overview of the configuration file options (structure overview)
+The Bao hypervisor's configuration is managed through a dedicated configuration
+in the form of a C source file. This section provides an in-depth
+description of the various configuration options available.
+
+The configuration file follows a specific structure, as outlined below:
+
+.. code-block:: c
+
+    #include <config.h>
+
+    // Load guests' image
+    VM_IMAGE(img1_name, "/path/to/vm1/binary.bin");
+    VM_IMAGE(img2_name, "/path/to/vm2/binary.bin");
+
+    struct config config = {
+
+        // Shared memories configuration
+        .shmemlist_size = N,
+        .shmemlist = (struct shmem[]) {
+            [0] = {/*shared memory config*/,},
+            [1] = {/*shared memory config*/,},
+            ...
+            [N] = {/*shared memory config*/,}
+        },
+
+        // Guests Configuration
+        .vmlist_size = NUM_VMs,
+        .vmlist = {
+            { /* VM 0 Config*/},
+            { /* VM 1 Config*/},
+            ...
+            { /* VM N Config*/},
+        }
+    };
+
+Before defining the configuration struct, it's crucial to declare the VM
+images. The configuration file anticipates receiving a compiled image of each
+guest (.bin file) which will then be allocated to individual VMs using the
```

Comment:
```suggestion
guest (``.bin`` file) which will then be allocated to individual VMs using the
```

Reply:
Nice catch, thanks! Applied the changes.